src/controllers/midi/midicontroller.cpp

```diff
@@ -13,6 +13,16 @@
 #include "moc_midicontroller.cpp"
 #include "util/math.h"
 
+MidiInputHandlerController::MidiInputHandlerController(
+        const LegacyMidiControllerMapping& mapping,
+        const MidiInputMapping& inputMapping)
+        : m_mapping(mapping), m_inputMapping(inputMapping) {
+}
+
+void MidiInputHandlerController::disconnect() {
+    m_mapping.removeInputMapping(this->m_inputMapping.key.key);
```

Comment:
In the long term, `removeInputMapping` probably needs a little more work because it just removes all mappings with the same key. I can imagine that it's probably not super uncommon that people will want to bind different handlers to the same key (midi status + control) and then only remove one of them. At least that's required to make a proper `MidiInputHandlerController.disconnect()` work.